draft.md

```diff
@@ -0,0 +1,81 @@
+# Draft Ipld Unixfs Spec
+
+## Basic Structure
+
+   - Some sort of header that indicates that this a directory and included a version number.  The header could also have fields to give additional information on the meaning of the extended attributes.
+
+  - CBOR Map
+    - Key: CBOR Byte or Text String: File Name
+    - Value: CBOR Array of:
+      - Type: CBOR Unsigned Int
+      - Link or Data: CBOR Type varies
+      - Optional file size: CBOR Unsigned Int
+      - Optional Standard Attributes: CBOR Map
+      - Optional Extended Attributes: CBOR Map
+
+The file size is only defined for regular files and is the size of the file contents.
+
+All maps should be ordered based on the binary values of the key,
+duplicates are not allowed.
+
+### Notes
+
+ * An array makes sense to be as this is more compact and the value of
+   the fields are unambiguous, it also allows for a separation of
+   standard and extended attributes
+
+ * The key type can either be a byte or text string as POSIX makes no
+   requirements that file names be utf-8 and it is important that any
+   file name can be faithfully represented, if the string is utf-8
+   then the type will be Text.
```

Comment:
This makes an implicit decision regarding the question I posed at the end of https://github.com/ipfs/go-ipfs/issues/4292#issuecomment-338181608: we shift the onus of "check that the name is safe to use/dipslay" to the consumers. Are we ready to do that?

Reply:
I think it is important to represent any valid file in a POSIX system.  I am agent restricting the range in the spec.  Yes I think it is the consumer job to make sure that the filename is safe to display.